main/store/migrate/index.ts

```diff
@@ -4,33 +4,28 @@ import legacyMigrations from './migrations/legacy'
 import migration35 from './migrations/35'
 import migration36 from './migrations/36'
 
-import type { Migration } from './types'
-
-const migrations: Record<number, Migration> = {
-  ...legacyMigrations,
-  ...Object.fromEntries([
-    [35, migration35],
-    [36, migration36]
-  ])
-}
+import type { Migration } from '../state'
+
+const migrations: Migration[] = [...legacyMigrations, migration35, migration36].sort(
+  (m1, m2) => m1.version - m2.version
+)
 
 // Version number of latest known migration
-const latest = Math.max(...Object.keys(migrations).map((n) => parseInt(n)))
+const latest = migrations[migrations.length - 1].version
```

Comment:
You might use dynamic import to avoid having to specify new migrations here each time, something like

1. get array of dirNames in migrations dir
2. map array to dynamic import of each migration index file
3. sort as on line 9 above

Reply:
I thought about this but deferred as it would make the migration process async and I thought it might be too much of a moving target to change the sync/async nature of the entire startup flow on top of all of these other changes. I do think this makes sense as a future optimization though as it will make it easier to manage migrations and not have to worry about adding them in two places.

Reply:
gonna just try a couple things though to see if there's something easy here

Reply:
ahh yeah I think we'd need to use top-level await and that would require us changing our TS config, so at a minimum I think this is best left for another PR